File: leetcode/Length_of_Longest_Subarray_With_at_Most_K_Frequency.cpp
#include <map>
class Solution
{
public:
    int maxSubarrayLength(vector<int> &nums, int k)
    {
        map<int, int> data;
        int ans = 0;
        int l = 0;
        for (int r = 0; r < nums.size(); r++)
        {
            data[nums[r]]++;
            if (data[nums[r]] > k)
            {
                while (data[nums[l]] != data[nums[r]])
                {
                    data[nums[l]]--;
                    l++;
                }
                data[nums[l]]--;
                l++;
                if (ans < r-l+1){}
            }
            ans = std::max(ans, r-l+1);
        }
        return ans;
    }
};

File: leetcode/March 2024/Length_of_Longest_Subarray_With_at_Most_K_Frequency.py
class Solution:
    def maxSubarrayLength(self, nums, k: int) -> int:
        ans = 0
        data = {}
        l = 0
        for r in range(len(nums)):
            data[nums[r]] = data.get(nums[r], 0) + 1
            if data[nums[r]] > k:
                while nums[l] != nums[r]:
                    data[nums[l]] -= 1
                    l += 1
                data[nums[l]] -= 1
                l += 1
            ans = max(ans, r - l + 1)
        return ans
